browser/extensions/brave_ipfs_client_updater.cc

```diff
@@ -0,0 +1,102 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/extensions/brave_ipfs_client_updater.h"
+
+#include "base/files/file_enumerator.h"
+#include "base/files/file_path.h"
+#include "base/files/file_util.h"
+#include "base/task/post_task.h"
+#include "third_party/re2/src/re2/re2.h"
+
+namespace extensions {
+
+std::string BraveIpfsClientUpdater::g_ipfs_client_component_id_(
+    kIpfsClientComponentId);
+std::string BraveIpfsClientUpdater::g_ipfs_client_component_base64_public_key_(
+    kIpfsClientComponentBase64PublicKey);
+
+BraveIpfsClientUpdater::BraveIpfsClientUpdater()
+    : task_runner_(
+          base::CreateSequencedTaskRunnerWithTraits({base::MayBlock()})),
+      registered_(false) {
+}
+
+BraveIpfsClientUpdater::~BraveIpfsClientUpdater() {
+}
+
+void BraveIpfsClientUpdater::Register() {
+  if (registered_)
+    return;
+
+  BraveComponentExtension::Register(kIpfsClientComponentName,
+                                    g_ipfs_client_component_id_,
+                                    g_ipfs_client_component_base64_public_key_);
+  registered_ = true;
+}
+
+base::FilePath BraveIpfsClientUpdater::GetExecutablePath() const {
+  return executable_path_;
+}
+
+void BraveIpfsClientUpdater::InitExecutablePath(
+    const base::FilePath& install_dir) {
+  base::FilePath executable_path;
+  base::FileEnumerator traversal(install_dir, false,
+                                 base::FileEnumerator::FILES,
+                                 FILE_PATH_LITERAL("ipfs-*"));
+  for (base::FilePath current = traversal.Next(); !current.empty();
+       current = traversal.Next()) {
+    base::FileEnumerator::FileInfo file_info = traversal.GetInfo();
+    if (!RE2::FullMatch(file_info.GetName().MaybeAsASCII(),
```

Comment:
We shouldn't use a wildcard here to find files (I know the Tor client updater does that, but we're planning to correct that soon). Instead, we should look for a specific file ("ipfs-client" or something similarly generic) and match the contents against a checksum.

Reply:
Also, note that the file must not have a ".exe" extension or the CRX unzipper will omit it.